pkg/util/kubernetes/clustername/clustername.go

```diff
@@ -108,11 +108,18 @@ func getClusterName(ctx context.Context, data *clusterNameData, hostname string)
 			}
 		}
 
-		if data.clusterName == "" && config.IsFeaturePresent(config.Kubernetes) {
-			clusterName, err := hostinfo.GetNodeClusterNameLabel(ctx)
+		if config.IsFeaturePresent(config.Kubernetes) {
```

Comment:
I think it's not exactly the logic we want. I think we need to add a return parameter from `GetNodeClusterNameLabel` to know if the retrieved value is from explicit label (e.g. custom label or generic DD label) or not.

Then we use the value returned if `clusterName == " || clusterName != "" && fromExplicitLabel`